substrate/frame/nomination-pools/src/migration.rs

```diff
@@ -118,51 +118,66 @@ pub mod unversioned {
 	/// sequence and hence can be applied unversioned on a production runtime.
 	pub struct DelegationStakeMigration<T>(sp_std::marker::PhantomData<T>);
 
+	/// We iterate over a maximum of `MAX_POOLS_BOUND` pools in a single block. At the time of
+	/// migration, there are currently less than `MAX_POOLS_BOUND` pools in the system. But if
+	/// there are more pools in the future, the remaining pools can be migrated via the
+	/// permission-less extrinsic [`Call::migrate_pool_to_delegate_stake`].
+	const MAX_POOLS_BOUND: u32 = 250;
```

Comment:
sounds good. We can even make this a parameter of `DelegationStakeMigration`. Polkadot has 230 pools now, so pretty close.